x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -687,6 +689,15 @@ static void writeDissect(PlanStreamOutput out, Dissect dissect) throws IOExcepti
         writeAttributes(out, dissect.extractedFields());
     }
 
+    static EmptyEsRelation readEmptyEsRelation(PlanStreamInput in) throws IOException {
+        return new EmptyEsRelation(readEsIndex(in));
+    }
+
+    static void writeEmptyEsRelation(PlanStreamOutput out, EmptyEsRelation relation) throws IOException {
```

Comment:
(sorry, the comment got lost for some reason... re-writing it here)
It's still not clear to me how these changes impact BWC, in particular with regards to Serverless with a mix of 8.11.1 and 8.11.2 nodes.
Nodes < 8.11.2 won't be able to deserialize EmptyEsRelation, maybe we should just we avoid to serialize them